ir/src/constraint_builder/evaluators.rs

```diff
@@ -52,11 +54,25 @@ impl ConstraintBuilder {
             // extend the constraint graph with all of the nodes from the evaluator graph
             self.graph.extend(ev_call_graph);
 
+            if constraints.len() > self.integrity_constraints.len() {
+                self.integrity_constraints
+                    .resize_with(constraints.len(), Vec::new);
+            }
+
             // add the roots of the evaluator function call's constraints
-            self.integrity_constraints
-                .resize_with(constraints.len(), Vec::new);
             for (segment, segment_constraints) in constraints.into_iter().enumerate() {
-                self.integrity_constraints[segment].extend(segment_constraints);
+                if let Some(selectors) = selectors {
+                    for constraint in segment_constraints {
+                        let constraint_with_selectors = self
+                            .insert_graph_node(Operation::Mul(*constraint.node_index(), selectors));
+                        self.integrity_constraints[segment].push(ConstraintRoot::new(
+                            constraint_with_selectors,
+                            constraint.domain(),
```

Comment:
This domain could be wrong, for example if the constraint's domain was previously `EveryRow` but one of the selectors is in the next row. We need to provide the domain for the newly created graph node